spec/scheduling-tasks.md

```diff
@@ -0,0 +1,380 @@
+
+<pre class='idl'>
+  enum TaskPriority {
+      "user-blocking",
+      "user-visible",
+      "background"
+  };
+</pre>
+
+{{TaskPriority/user-blocking}} is the highest priority, and is meant to be used
+for tasks that are blocking the user's ability to interact with the page, such
+as rendering the core experience or responding to user input.
+
+{{TaskPriority/user-visible}} is the second highest priority, and is meant to
+be used for tasks that visible to the user but not necessarily blocking user
+actions, such as rendering secondary parts of the page. This is the default
+priority.
+
+{{TaskPriority/background}} is the lowest priority, and is meant to be used for
+tasks that are not time-critical, such as background log processing or
+initializing certain third party libraries.
+
+Note: Tasks scheduled through a given {{Scheduler}} run in *strict priority
+order*, meaning the scheduler will always run {{TaskPriority/user-blocking}}
+tasks before {{TaskPriority/user-visible}} tasks, which in turn always run
+before {{TaskPriority/background}} tasks.
+
+<div algorithm>
+  {{TaskPriority}} |p1| is <dfn for="TaskPriority">less than</dfn>
+  {{TaskPriority}} |p2| if |p1| is less than |p2| in the
+  following total ordering:
+    * {{TaskPriority/background}} < {{TaskPriority/user-visible}} < {{TaskPriority/user-blocking}}
+</div>
+
+
+The `Scheduler` Interface {#sec-scheduler}
+---------------------
+
+Developers interact with the {{Scheduler}} interface to schedule tasks, using
+the {{Scheduler/postTask()}} method. A task consists of a callback, which is the
+entrypoint to the task, and 0 or more {{SchedulerPostTaskOptions}}:
+
+ - {{SchedulerPostTaskOptions/priority}}: An optional {{TaskPriority}}. If set,
+   this priority will be used to schedule the task, and the task's priority is
+   immutable. If left unspecified, the {{SchedulerPostTaskOptions/signal}}
+   option will determine the task's priority if set, otherwise the priority
+   defaults to {{TaskPriority/user-visible}}.
+
+ - {{SchedulerPostTaskOptions/signal}}: An optional {{AbortSignal}} or
+   {{TaskSignal}}. If a {{TaskSignal}} is specified and the
+   {{SchedulerPostTaskOptions/priority}} is ommitted, the signal is used to
+   determine the priority, which can be modified by the associated
+   {{TaskController}}. The {{SchedulerPostTaskOptions/signal}} is also used to
+   cancel pending tasks. See [Controlling Tasks](#sec-controlling-tasks) for
+   details on using a {{TaskController}} or {{AbortController}} with
+   {{Scheduler/postTask()}}.
+
+ - {{SchedulerPostTaskOptions/delay}}: An optional delay (milliseconds) may be
+   specified in order to delay the execution of the task.
+
+{{Scheduler/postTask()}} returns a Promise that is fulfilled with the result of
+the callback, or rejected with an "{{AbortError!!exception}}" {{DOMException}}
+if the task is aborted.
+
+<xmp class='idl'>
+  dictionary SchedulerPostTaskOptions {
+      AbortSignal? signal = null;
```

Comment:
Don't make dictionary members nullable. Instead just leave them as their default of non-required. This has slight observable effects in that `{ signal: null }` or `{ priority: null }` will throw, which is more idiomatic.